golang/internal/grpc/grpc.go

```diff
@@ -286,25 +270,73 @@ func grpcLoop(
 			}
 		}
 
-		command := new(agent.AgentCommand)
-		err = stream.RecvMsg(command)
-		if err != nil {
-			grpcConn.Client = nil
-			health.SetHealthGRPCStatus(false)
+		if stream != nil {
+			command := new(agent.AgentCommand)
+			err = stream.RecvMsg(command)
+			if err != nil {
+				grpcConn.Client = nil
+				health.SetHealthGRPCStatus(false)
 
-			if err == io.EOF {
-				log.Info().Msg("End of stream")
-			} else {
-				log.Error().Stack().Err(err).Msg("Cannot receive stream")
-				// TODO replace the line above with an error status code check and terminate dagent accordingly
+				if err == io.EOF {
+					log.Info().Msg("End of stream")
+				} else {
+					log.Error().Stack().Err(err).Msg("Cannot receive stream")
+					break
+				}
+				time.Sleep(minimumRPCJitterMillis * time.Millisecond)
 			}
 
-			time.Sleep(appConfig.DefaultTimeout)
-			continue
+			stop, err := grpcProcessCommand(ctx, workerFuncs, command, cancel, appConfig)
```

Comment:
If the stream ends, after an EOF this will still process the empty command which should return an ErrUnkownAgentCommand, I think. But after an EOF grpcLoop should just return, no?